source/Calamari/Kubernetes/SetupKubectlAuthentication.cs

```diff
@@ -50,12 +52,22 @@ public CommandResult Execute(string accountType)
                 environmentVars[proxyVariable.Key] = proxyVariable.Value;
             }
 
-            var kubeConfig = CreateKubectlConfig();
+            // Make sure Kubectl exists and can be run
             if (!kubectl.TrySetKubectl())
             {
                 return errorResult;
             }
 
+            // If we are running on a Kubernetes Tentacle Target,
+            // then we can rely on using the service account inside the pod for authentication.
+            if (variables.Get("Octopus.Machine.DeploymentTargetType") == KubernetesTentacleTargetTypeId)
```

Comment:
@APErebus I've opted to just check if the current Deployment Target Type is the KubernetesTentacle. Does that sound reasonable to you?